channels/tsmf/client/alsa/tsmf_alsa.c

```diff
@@ -68,8 +68,7 @@ static BOOL tsmf_alsa_open(ITSMFAudioDevice *audio, const char *device)
 	TSMFAlsaAudioDevice *alsa = (TSMFAlsaAudioDevice *) audio;
 	if(!device)
 	{
-		if(!alsa->device[0])
-			strncpy(alsa->device, "default", sizeof(alsa->device));
+		strncpy(alsa->device, "default", sizeof(alsa->device));
 	}
```

Comment:
No.
freerdp_tsmf_client_audio_subsystem_entry() -> tsmf_alsa_open() ... tsmf_alsa_free()
If we call  tsmf_alsa_open() more than once than we have resource leak.
rdpsnd have Close() and reopen work by design.
tsmf does not have Close() and device can not be changed.
In tsmf OSS code i add checks to prevent reopen.
ALSA code is weird, especially in rdpsnd.